crates/holochain_types/src/warrant.rs

```diff
@@ -17,60 +15,42 @@ use crate::dht_op::DhtOpUniqueForm;
     Eq,
     PartialEq,
     Hash,
-    derive_more::Constructor,
+    derive_more::From,
+    derive_more::Deref,
 )]
 #[cfg_attr(
     feature = "fuzzing",
     derive(arbitrary::Arbitrary, proptest_derive::Arbitrary)
 )]
-pub struct WarrantOp {
-    /// The warrant which was issued
-    pub warrant: Warrant,
-    /// author of the warrant
-    pub author: AgentPubKey,
-    /// signature of (Warrant, Timestamp) by the author
-    pub signature: Signature,
-    /// time when the warrant was issued
-    pub timestamp: Timestamp,
-}
+pub struct WarrantOp(SignedWarrant);
 
 impl WarrantOp {
     /// Get the type of warrant
     pub fn get_type(&self) -> WarrantOpType {
-        match self.warrant {
-            Warrant::ChainIntegrity(_) => WarrantOpType::ChainIntegrityWarrant,
+        match self.proof {
+            WarrantProof::ChainIntegrity(_) => WarrantOpType::ChainIntegrityWarrant,
         }
     }
 
-    /// Convert the WarrantOp into a SignedWarrant
-    pub fn into_signed_warrant(self) -> SignedWarrant {
-        Signed::new((self.warrant, self.timestamp).into(), self.signature)
+    /// Sign the warrant for use as an Op
```

Comment:
```suggestion
    /// Sign the warrant for use as a WarrantOp
```

The name of this function confuses me a little. I think it means 'I've constructed the warrant and I'm ready to author it' but from elsewhere it looks like a getter to retrieve the `author` from the warrant op. A more descriptive name might help here?